components/formats-gpl/src/loci/formats/in/CellomicsReader.java

```diff
@@ -215,6 +215,36 @@ protected void initFile(String id) throws FormatException, IOException {
     if (fields * wellRows * wellColumns > files.length) {
       files = new String[] {id};
     }
+    final int wellColumnsFinal = wellColumns;
+    Arrays.sort(files, new Comparator<String>() {
+        @Override
+        public int compare(String f1, String f2) {
+            int wellRow1 = getWellRow(f1);
+            int wellCol1 = getWellColumn(f1);
+            int field1_1 = getField(f1);
+            String field1 = String.valueOf(field1_1);
+            if (field1_1<10) {
+                field1 = "00" + field1;
+            }else if (field1_1<100) {
+                field1 = "0" + field1;
+            }
+
+            int wellRow2 = getWellRow(f2);
+            int wellCol2 = getWellColumn(f2);
+            int field2_2 = getField(f2);
+
+            String field2 = String.valueOf(field2_2);
+            if (field2_2<10) {
+                field2 = "00" + field2;
+            }else if (field2_2<100) {
+                field2 = "0" + field2;
+            }
+
+            int fileId1 = Integer.valueOf(String.valueOf((wellRow1 * wellColumnsFinal) + wellCol1) + field1);
+            int fileId2 = Integer.valueOf(String.valueOf((wellRow2 * wellColumnsFinal) + wellCol2) + field2);
+            return fileId1 - fileId2;
```

Comment:
It shouldn't be necessary to convert the well and field to a new string for comparison.  The values of ```wellRow*```, ```wellCol*``` and ```field*_*``` can be compared directly, similar to https://github.com/openmicroscopy/bioformats/blob/develop/components/formats-gpl/src/loci/formats/in/ScanrReader.java#L460.  This would have the added advantage of not needing to be updated if we encounter a dataset with more than 1000 fields.